pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -313,10 +313,7 @@ func (p *PriorityQueue) AddUnschedulableIfNotPresent(pod *v1.Pod, podSchedulingC
 	// it to unschedulableQ.
 	if p.moveRequestCycle >= podSchedulingCycle {
 		if err := p.podBackoffQ.Add(pInfo); err != nil {
-			// TODO: Delete this klog call and log returned errors at the call site.
-			err = fmt.Errorf("error adding pod %v to the backoff queue: %v", pod.Name, err)
-			klog.Error(err)
-			return err
+			return fmt.Errorf("error adding pod %v to the backoff queue: %v", pod.Name, err)
```

Comment:
A nit: as you wrap the detailed msg here as an `err`, and re-wrap the err msg in caller side, I suppose we'll get a sort of duplicate err msg? Like:

```
Error add unschedulable pod <pod> to queue: error adding pod <pod> to the backoff queue: <err>.
```

Maybe it's better wrapping the detailed msg only in one side - (1) either wrap here and in caller side simply log `klog.Errorf(err)`, or (2) simply return err here and in caller side log `klog.Errorf("Error add unschedulable pod %v to queue: %v", pod, err)`.

(1) is more preferable IMO.

Reply:
Yeah, that's a good one. Updated the code in caller site with `klog.Errorf(err)`. Please take a look at the latest commit.